internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -2448,56 +2451,222 @@ func TestComputeDesiredMachine(t *testing.T) {
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					MachineNamingStrategy: &clusterv1.MachineNamingStrategy{
+						Template: "some-hardcoded-name-{{ .doesnotexistindata }}-{{ .random }}", // invalid template
+					},
+					Template: machineTemplateSpec,
+				},
+			},
+			existingMachine: nil,
+			wantMachine:     nil,
+		},
 		{
-			name:            "creating a new Machine",
+			name: "should return the correct Machine object when creating a new Machine with default templated name",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					Template: machineTemplateSpec,
+				},
+			},
 			existingMachine: nil,
-			want:            expectedNewMachine,
+			wantMachine:     expectedNewMachine,
+			wantName: []gomegatypes.GomegaMatcher{
+				HavePrefix(msName),
+			},
 		},
 		{
-			name:            "updating an existing Machine",
+			name: "updating an existing Machine",
+			ms: &clusterv1.MachineSet{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: metav1.NamespaceDefault,
+					Name:      msName,
+					Labels: map[string]string{
+						clusterv1.MachineDeploymentNameLabel: mdName,
+					},
+				},
+				Spec: clusterv1.MachineSetSpec{
+					ClusterName:     testClusterName,
+					Replicas:        ptr.To[int32](3),
+					MinReadySeconds: 10,
+					Selector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"k1": "v1"},
+					},
+					Template: machineTemplateSpec,
+				},
+			},
 			existingMachine: existingMachine,
-			want:            expectedUpdatedMachine,
+			wantMachine:     expectedUpdatedMachine,
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
-			got := (&Reconciler{}).computeDesiredMachine(ms, tt.existingMachine)
-			assertMachine(g, got, tt.want)
+			var got *clusterv1.Machine
+			var err error
+			msr := &Reconciler{
+				Client: fake.NewClientBuilder().WithObjects(
+					testCluster,
+					tt.ms,
+				).WithStatusSubresource(&clusterv1.MachineSet{}).Build(),
+				recorder: record.NewFakeRecorder(32),
+			}
+			got, err = msr.computeDesiredMachine(tt.ms, tt.existingMachine)
+
+			if tt.wantMachine == nil {
+				g.Expect(err).To(HaveOccurred())
+				return
+			}
+			assertMachine(g, got, tt)
 		})
 	}
 }
 
-func assertMachine(g *WithT, actualMachine *clusterv1.Machine, expectedMachine *clusterv1.Machine) {
+func assertMachine(g *WithT, actualMachine *clusterv1.Machine, testData computeDesiredMachineTestCase) {
```

Comment:
Nit: I'd try to keep this simple and instead of passing an arbitrary struct which contains more information than needed, just pass in the additonal stuff.


```suggestion
func assertMachine(g *WithT, actualMachine *clusterv1.Machine, expectedMachine *clusterv1.Machine, nameMatches []gomegatypes.GomegaMatcher) {
```

Reply:
done